osu.Game/Graphics/UserInterface/OsuModSettingsTextBox.cs

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Allocation;
+using osu.Framework.Extensions.Color4Extensions;
+using osu.Framework.Graphics.Colour;
+using osu.Framework.Input.Events;
+
+namespace osu.Game.Graphics.UserInterface
+{
+    public class OsuModSettingsTextBox : OsuTextBox
+    {
+        private const float border_thickness = 3;
+
+        private SRGBColour borderColourFocused;
+        private SRGBColour borderColourUnfocused;
+
+        [BackgroundDependencyLoader]
+        private void load(OsuColour colour)
+        {
+            borderColourUnfocused = colour.Gray4.Opacity(0.5f);
+            borderColourFocused = BorderColour;
+
+            BorderThickness = border_thickness;
+            BorderColour = borderColourUnfocused;
+        }
+
+        protected override void OnFocus(FocusEvent e)
+        {
+            base.OnFocus(e);
+
+            BorderThickness = border_thickness;
+            BorderColour = borderColourFocused;
```

Comment:
Not sure why `BorderThickness` is assigned the same value in three places, and the `BorderColour` update can be extracted to a separate `updateState()` method which would select the colour based on the value of `HasFocus`.

Reply:
`BorderThickness` has to assigned in `load()` and `OnFocusLost()` because the default value is `0` by default and it's set to `0` in `base.OnFocusLost()`. It doesn't necessarily need to be assigned in `OnFocus()`, but I moved the assignment into `updateBorder()` anyway.